flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -260,6 +260,18 @@ void releaseAllResources() throws IOException {
 		}
 	}
 
+	int unSafeGetSizeOfQueuedBuffer() {
```

Comment:
Can we extract this and `RemoteInputChannel#unsynchronizedGetNumberOfQueuedBuffers` to `InputChannel` interface? Thanks to that, we will not need to `instance of` check in the `SingleInputGate` + can you add test coverage for this in `LocalInputChannelTest`?